pandas/_libs/reduction.pyx

```diff
@@ -48,7 +48,7 @@ cdef class _BaseGrouper:
             # See the comment in indexes/base.py about _index_data.
             # We need this for EA-backed indexes that have a reference
             # to a 1-d ndarray like datetime / timedelta / period.
-            object.__setattr__(cached_ityp, '_index_data', islider.buf)
+            object.__setattr__(cached_ityp, '_data', islider.buf)
```

Comment:
I'm not sure of the distinction between `_data` and `_index_data` so this might be incorrect, but did help reduce a lot of failures.

cc @TomAugspurger if you have any insights

Reply:
if this works and we can get rid of _index_data thatd be great

Reply:
Would you happen to know if the original intent is that _data can be an EA but _index_data must always be an ndarray?

Reply:
> Would you happen to know if the original intent is that _data can be an EA but _index_data must always be an ndarray?

I think that's right.